### ISSUE TITLE
"secureConnection" value in config not working with Mailgun SMTP

### ISSUE THREAD

**Issue by iamtk** [bitnami/bitnami-docker-ghost#97]
When I populate the mail environment variables using Mailgun's SMTP details, having the below line:

`"secureConnection": "true",`

or

`"secureConnection": "false",`

Causes emails not to send.  Removing that line completely from the 'config.production.json' file and restarting Ghost fixes the problem.

Example:

```json
"mail": {
    "from": "'Example' <noreply@example.com>",
    "transport": "SMTP",
    "options": {
      "host": "smtp.mailgun.org",
      "port": "587",
      "secureConnection": "false",
      "auth": {
        "user": "mymailgunuser@example.com",
        "pass": "password"
      },
      "service": "Mailgun"
    }
  }
```
Experienced this using images 3.40.0 through 3.40.2 (possibly before also I haven't checked).  I'm running Docker version 19.03.13, on Ubuntu 20.04.1 LTS.

This line has been created since being introduced as a ENV variable in the dockerfile, is there a way to override?  Omitting or setting to -e SMTP_SECURE_CONNECTION="" just uses the default ENV value for this of false.

Apologies if I am missing some info that's needed, please let me know and I will update.

**Comment by rafariossaa** [comment]
Hi,
  I think the issue here may be that the value for `secureConnection` is quoted and [here in the documentation](https://ghost.org/docs/config/#secure-connection) it is not.
  Could you try by setting `"secureConnection": false` ?

**Comment by iamtk** [comment]
I should have spotted that.  Yes putting the boolean value without quotes ("secureConnection": false) - it works fine.

I'm guessing this JSON is generated by the Bitnami image because of it's ENV variables so is it possible to get this fixed to match the documentation so we don't have to manually correct (remove the quotes) it each time a new instance is created?

**Comment by rafariossaa** [comment]
Yes, this is an issue in the image.
We are fixing it.

**Comment by rafariossaa** [comment]
Hi,
  Sorry for the long delay, could you give it a try again ?

**Comment by Bluefitdev** [comment]
@rafariossaa , when I tried to put it in [bitnami k8s ghost](https://github.com/bitnami/charts/blob/master/bitnami/ghost/values.yaml), I can't put it without quote. It will throw me an error

```
extraEnvVars:
  - name: SMTP_SECURE_CONNECTION
     value: true  <---- without quote, installing helm chart will throw error. Is this expected?
```

**Comment by rafariossaa** [comment]
Hi,
  Can you provide the error log ?

**Comment by Bluefitdev** [comment]
There's no error log actually, it just doesn't compile if i put boolean value in the `extraEnvVars`. it shows all of my other variables like a corrupt yaml file.

**Comment by iamtk** [comment]
> Hi,
> Sorry for the long delay, could you give it a try again ?

Creating a docker service using this image with -e SMTP_SECURE_CONNECTION=false still (unless I have missed something) results in "secureConnection": "false" (boolean value with quotes) being created in the config.production.json file (causing emails not to send).  Is this not because when the variable is mentioned in this file here - rootfs/ghost-inputs.json - it still has quotes?

**Comment by rafariossaa** [comment]
Hi,
  There is an issue in the container when rendering the config file. The value from the environment variable is handled always as a string, hence it is not being rebdered correctly in the config file.
  We have a internal task to fix this.

**Comment by juan131** [comment]
Hi @iamtk 

We just released a new revision of the container (`4.8.4-debian-10-r7`) that changes the logic of the container so it's based on Bash, find more information in the link below:

- https://github.com/bitnami/bitnami-docker-ghost#3425-debian-10-r67-and-484-debian-10-r7

As you can see at the lines below, this new logic uses `jq` to write/read in the Ghost configuration file and it's able to parse correctly strings, binaries and booleans:

- https://github.com/bitnami/bitnami-docker-ghost/blob/master/4/debian-10/rootfs/opt/bitnami/scripts/libghost.sh#L296
- https://github.com/bitnami/bitnami-docker-ghost/blob/master/4/debian-10/rootfs/opt/bitnami/scripts/libghost.sh#L188

Please also note this new container doesn't use the `SMTP_SECURE_CONNECTION` env. variable anymore. Instead set the `GHOST_SMTP_PROTOCOL` env. variable to `ssl` or `tls` to enable secure connections.

> The reasons for these changes are related to standardising the env. variables we expose in Bitnami containers to configure SMTP.

Please give a try and do not hesitate to reopen the issue if you continue facing issues.
